File: backend/app/db/models/policies.py
from __future__ import annotations
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class PolicyRow(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    organization_id: UUID
    name: str
    category: str | None = None
    status: str = "draft"
    owner_user_id: UUID | None = None


class PolicyVersionRow(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    policy_id: UUID
    version_label: str
    content: str
    effective_from: datetime | None = None
    effective_to: datetime | None = None
    status: str = "draft"
    created_by: UUID | None = None
    approved_by: UUID | None = None
    hindsight_document_id: str | None = None
    created_at: datetime

File: frontend/src/lib/types/policies.ts
import type { PolicyStatus } from './common'

export interface PolicyVersion {
  id: string
  policy_id: string
  version_label: string
  content: string
  effective_from?: string | null
  effective_to?: string | null
  status: PolicyStatus
  created_at: string
  [key: string]: unknown
}

export interface Policy {
  id: string
  organization_id: string
  name: string
  category?: string | null
  status: PolicyStatus
  owner_user_id?: string | null
  versions: PolicyVersion[]
  [key: string]: unknown
}

export interface PolicyCreate {
  name: string
  category?: string
  content: string
  effective_from?: string
}

export interface PolicyUpdate {
  name?: string
  category?: string
  status?: PolicyStatus
}

export interface PolicyListParams {
  page?: number
  page_size?: number
}
